Mathlib/Algebra/Category/ModuleCat/Presheaf/OverColimitsRing'.lean

```diff
@@ -0,0 +1,304 @@
+import Mathlib.Algebra.Category.Ring.FilteredColimits
+import Mathlib.Algebra.Category.Ring.Colimits
+import Mathlib.Algebra.Category.Grp.FilteredColimits
+import Mathlib.Algebra.Category.ModuleCat.Colimits
+import Mathlib.Algebra.Category.ModuleCat.Presheaf
+import Mathlib.CategoryTheory.Limits.ConcreteCategory
+
+open CategoryTheory Category Limits Opposite
+
+universe u v
+
+section
+
+
+variable {C : Type u} [Category.{u} C] [IsCofiltered C]
+variable {ring : Type u} [Category.{u+1} ring] [ConcreteCategory.{u+1} ring]
+variable {ab : Type u} [Category.{u+1} ab] [ConcreteCategory.{u+1} ab]
+attribute [local instance] ConcreteCategory.hasCoeToSort
+attribute [local instance] ConcreteCategory.instFunLike
+
+variable [∀ x : ring, Ring x] [∀ x : ab, AddCommGroup x]
+variable [∀ x y : ring, RingHomClass (x ⟶ y) x y]
+variable [∀ x y : ab, AddMonoidHomClass (x ⟶ y) x y]
+variable (R : Cᵒᵖ ⥤ ring) (M : Cᵒᵖ ⥤ ab)
+variable [HasColimit R] [HasColimit M]
+variable [PreservesColimit R (forget ring)] [PreservesColimit M (forget ab)]
+variable [∀ c, Module (R.obj c) (M.obj c)]
+variable (compatible_smul : ∀ {c₁ c₂ : Cᵒᵖ} (i₁ : c₁ ⟶ c₂) (r : R.obj $ c₁) (m : M.obj $ c₁),
+    M.map i₁ (r • m) = R.map i₁ r • M.map i₁ m)
```

Comment:
This is another approach, it works for `RingCat` and `CommRingCat` simultaneously.

Reminder to myself: change `C\op` to `C`